novobi_sales_b2b/models/sale_request.py

```diff
@@ -0,0 +1,166 @@
+class SalesRequest(models.Model):
+    _name = 'sale.request'
+    _order = 'date_request desc, id desc'
+    _inherit = ['mail.thread', 'mail.activity.mixin']
+    _description = "Sales Request"
+    _check_company_auto = True
+
+    name = fields.Char(string='Request Reference', required=True, index=True, states={
+                       'draft': [('readonly', False)]}, default=lambda self: _('New'))
+    date_request = fields.Datetime(
+        string='Request Date', required=True, readonly=True, index=True, default=fields.Datetime.now)
+    date_confirm = fields.Datetime(
+        string='Confirm Date', readonly=True, index=True)
+    company_id = fields.Many2one(
+        'res.company', 'Company', required=True, index=True, default=lambda self: self.env.company)
+    sale_user_id = fields.Many2one(
+        'res.users', string='Salesperson', index=True, default=lambda self: self.env.user,
+        domain=lambda self: [('groups_id', 'in', self.env.ref('sales_team.group_sale_salesman').id)])
+    pricelist_id = fields.Many2one(
+        'product.pricelist', string='Pricelist', check_company=True, required=True,
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]")
+    currency_id = fields.Many2one(related='pricelist_id.currency_id', depends=[
+                                  "pricelist_id"], store=True, ondelete="restrict")
+    partner_id = fields.Many2one(
+        'res.partner', string='Retailer',
+        required=True, change_default=True, index=True)
+    request_state = fields.Selection([
+        ('pending', 'Pending'),
+        ('approved', 'Approved'),
+        ('rejected', 'Rejected'),
+        ('cancelled', 'Cancelled')
+    ], string='Request Status', default='pending')
+    fulfillment_state = fields.Selection([
+        ('ready', 'Ready'),
+        ('waiting', 'Waiting'),
+        ('done', 'Done')
+    ], string='Fulfillment Status', default='ready')
+    tag_ids = fields.Many2many(
+        'crm.tag', string='Tags')
+    order_id = fields.Many2one(
+        'sale.order', string='Sale order', ondelete='cascade')
+    request_line = fields.One2many(
+        'sale.request.line', 'request_id', string='Request Lines', ondelete='cascade')
+    amount_total = fields.Monetary(
+        string='Total', store=True, compute='_compute_amount_total_item')
+    amount_item = fields.Integer(
+        string='Items', store=True, compute='_compute_amount_total_item')
+
+    @api.depends('request_line', 'request_line.price_subtotal')
+    def _compute_amount_total_item(self):
+        """
+        Compute the total amounts, items of the RS.
+        """
+        for request in self:
+            amount_total = 0.0
+            amount_item = 0
+            for line in request.request_line:
+                amount_total += line.price_subtotal
+                amount_item += 1
+            request.update({
+                'amount_total': amount_total,
+                'amount_item': amount_item
+            })
```

Comment:
amount_item is a counter? why not use len(request.request_line)?
to calculate amount_total, we can use sum(request.request_line.mapped('price_subtotal'))